x-pack/plugins/actions/server/lib/task_runner_factory.ts

```diff
@@ -193,10 +183,95 @@ export class TaskRunnerFactory {
           }
         }
       },
+      cancel: async () => {
+        // Write event log entry
+        const actionTaskExecutorParams = taskInstance.params as ActionTaskExecutorParams;
```

Comment:
Would it make sense to store the `actionInfo` that is retrieved during `ActionExecutor.execute` in the `ActionExecutor` and then call a function like `ActionExecutor.logCancellation` that uses that internal variable to create the event log record? I guess there is the chance that something went wrong with the executor and that action info wasn't retrieved and isn't available, but in that case, we're just missing some fields from the event log entry vs re-retrieving the action info on cancellation. WDYT?

Reply:
Sounds great! I will do the changes.